src/ILLink.Shared/Dataflow/DictionaryLattice.cs

```diff
@@ -0,0 +1,77 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+
+namespace ILLink.Shared
+{
+	public class DefaultValueDictionary<TKey, TValue> : Dictionary<TKey, TValue>,
```

Comment:
Might be a good idea to move this to a separate file - it's used outside of the DictionaryLattice below. And it's a more "general use case" class I think.